### PR TITLE
fix t8n tool errors

### PR DESCRIPTION
**Pull request by gurukamath**
### What was wrong?
Some edge cases were not handled properly by t8n

Related to Issue #773 

### How was it fixed?
1. Fix bug to accept state fork `ConstantinopleFix` instead of `Constantinople`
2. Update evm tools tests to use `GeneralStateTests` fixtures
3. Handle invalid transactions without crashing the `t8n` tool.
4. Handle a couple of testing edge cases.
 
#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: src/ethereum/arrow_glacier/fork.py [hunk 1]
```diff
@@ -762,10 +763,13 @@ def process_transaction(
...
-    if isinstance(tx, FeeMarketTransaction):
-        gas_fee = tx.gas * tx.max_fee_per_gas
-    else:
-        gas_fee = tx.gas * tx.gas_price
+    try:
+        if isinstance(tx, FeeMarketTransaction):
+            gas_fee = tx.gas * tx.max_fee_per_gas
+        else:
+            gas_fee = tx.gas * tx.gas_price
+    except ValueError as e:
```
Comment (SamWilsn) [inline src/ethereum/arrow_glacier/fork.py:771]: What raises `ValueError` here? Integer overflow?

If it is an integer overflow, we should handle this _for all_ integer overflows, and not just in these specific cases. Should we change `base_types` so that it throws a specific `OverflowError` that subclasses `ValueError`?

File: src/ethereum_spec_tools/evm_tools/utils.py [hunk 2]
```diff
@@ -80,6 +85,8 @@ def get_module_name(forks: Any, options: Any, stdin: Any) -> Tuple[str, int]:
     """
     Get the module name and the fork block for the given state fork.
     """
+    if options.state_fork.upper() in UNSUPPORTED_FORKS:
```
Comment (SamWilsn) [inline src/ethereum_spec_tools/evm_tools/utils.py:88]: Should use [`str.casefold`](https://docs.python.org/3/library/stdtypes.html#str.casefold) for case-insensitive comparisons.

File: src/ethereum/arrow_glacier/fork.py (hunk 1 above)
Comment (gurukamath) [reply src/ethereum/arrow_glacier/fork.py:771]: > What raises `ValueError` here? Integer overflow?

Yes. This is due to overflows. This change has been made because the testing framework expects an `InvalidBlock` exception (as it should) but receives `ValueError`.


> If it is an integer overflow, we should handle this _for all_ integer overflows, and not just in these specific cases. Should we change `base_types` so that it throws a specific `OverflowError` that subclasses `ValueError`?

I did consider modifying the `base_types`, but raising an `InvalidBlock` exception (or a subclass of it) directly in the `base_types` does not seem right to me. (Although one could argue that the spec base types are used only in the context of processing blocks and so, an `InvalidBlock` exceptions is not out of place there). What do you think?

Alternatively, I also considered modifying the testing framework to handle the `ValueError` but that seems too generic of an exception to expect in the tests. Would be better to keep the expected exceptions as specific as possible.

File: src/ethereum_spec_tools/evm_tools/utils.py (hunk 2 above)
Comment (gurukamath) [reply src/ethereum_spec_tools/evm_tools/utils.py:88]: Will update.

File: src/ethereum/arrow_glacier/fork.py (hunk 1 above)
Comment (petertdavies) [reply src/ethereum/arrow_glacier/fork.py:771]: Another approach, which we take with gas tracking in the EVM, is to make everything a `Uint` instead of a `U256`. If we do that the overflow won't happen at all and instead `InvalidBlock` will get raised a few lines later.

Comment (gurukamath) [reply src/ethereum/arrow_glacier/fork.py]: Updated to use `Uint` for all gas.

**Review by SamWilsn — APPROVED** [review]
